developer_manual/basics/controllers.rst

```diff
@@ -222,15 +223,44 @@ To set, get or modify session variables, the ISession object has to be injected
 
 Nextcloud will read existing session data at the beginning of the request lifecycle and close the session afterwards. This means that in order to write to the session, the session has to be opened first. This is done implicitly when calling the set method, but would close immediately afterwards. To prevent this, the session has to be explicitly opened by calling the reopen method.
 
-Alternatively you can use the ``@UseSession`` annotation to automatically open and close the session for you.
+Alternatively, you can use the ``#[UseSession]`` attribute to automatically open and close the session for you.
+
+.. code-block:: php
+    :emphasize-lines: 2,13
+
+    use OCP\AppFramework\Controller;
+    use OCP\AppFramework\Http\Attribute\UseSession;
+    use OCP\AppFramework\Http\Response;
+
+    class PageController extends Controller {
+
+        #[UseSession]
+        public function writeASessionVariable(): Response {
+            // ...
+        }
+
+    }
+
+.. note:: The ``#[UseSession]`` was added in Nextcloud 26 and requires PHP8.0 or later. If your app targets older releases and PHP7.x then use the deprecated ``@UseSession`` annotation.
+
+    .. code-block:: php
+        :emphasize-lines: 2
+
+        /**
+         * @UseSession
+         */
+        public function writeASessionVariable(): Response {
+            // ....
+        }
+
 
 In case the session may be read and written by concurrent requests of your application, keeping the session open during your controller method execution may be required to ensure that the session is locked and no other request can write to the session at the same time. When reopening the session, the session data will also get updated with the latest changes from other requests. Using the annotation will keep the session lock for the whole duration of the controller method execution.
 
 For additional information on how session locking works in PHP see the artile about `PHP Session Locking: How To Prevent Sessions Blocking in PHP requests <https://ma.ttias.be/php-session-locking-prevent-sessions-blocking-in-requests/>`_.
```

Comment:
```suggestion
For additional information on how session locking works in PHP see the article about `PHP Session Locking: How To Prevent Sessions Blocking in PHP requests <https://ma.ttias.be/php-session-locking-prevent-sessions-blocking-in-requests/>`_.
```